src/Agent.Plugins/GitSourceProvider.cs

```diff
@@ -670,6 +670,15 @@ public string GenerateAuthHeader(AgentTaskPluginExecutionContext executionContex
 
             List<string> additionalFetchArgs = new List<string>();
             List<string> additionalLfsFetchArgs = new List<string>();
+
+            // Force Git to HTTP/1.1. Otherwise IIS will reject large pushes to Azure Repos due to the large content-length header
+            // This is caused by these header limits - https://docs.microsoft.com/en-us/iis/configuration/system.webserver/security/requestfiltering/requestlimits/headerlimits/
+            int exitCode_configHttp = await gitCommandManager.GitConfig(executionContext, targetPath, "http.version", "HTTP/1.1");
+            if (exitCode_configHttp != 0)
+            {
+                throw new InvalidOperationException($"Git config failed with exit code: {exitCode_configHttp}");
```

Comment:
Do we definitely want to fail due to this or just set a warning?

Reply:
Probably just a warning is fine, I've updated to do that. This should really never happen anyways though (it would probably mean something is pretty wrong with your git config).